pkg/controller/logstorage/elastic/elastic_controller.go

```diff
@@ -0,0 +1,710 @@
+	err = c.Watch(&source.Kind{Type: &storagev1.StorageClass{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch StorageClass resource: %w", err)
+	}
+
+	if err = c.Watch(&source.Kind{Type: &apps.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.ECKOperatorNamespace, Name: render.ECKOperatorName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch StatefulSet resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &esv1.Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.ElasticsearchNamespace, Name: render.ElasticsearchName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Elasticsearch resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &kbv1.Kibana{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.KibanaNamespace, Name: render.KibanaName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Kibana resource: %w", err)
+	}
+	if err = utils.AddSecretsWatch(c, render.TigeraElasticsearchInternalCertSecret, common.OperatorNamespace()); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Secret resource: %w", err)
+	}
+	if err = utils.AddSecretsWatch(c, render.TigeraElasticsearchInternalCertSecret, render.ElasticsearchNamespace); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Secret resource: %w", err)
+	}
+	if err = utils.AddConfigMapWatch(c, render.ECKLicenseConfigMapName, render.ECKOperatorNamespace, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch ConfigMap resource: %w", err)
+	}
+
+	for _, name := range []string{
+		render.OIDCUsersConfigMapName,
+		render.OIDCUsersEsSecreteName,
+		render.ElasticsearchAdminUserSecret,
+	} {
+		if err = utils.AddConfigMapWatch(c, name, render.ElasticsearchNamespace, &handler.EnqueueRequestForObject{}); err != nil {
+			return fmt.Errorf("log-storage-controller failed to watch ConfigMap resource: %w", err)
+		}
+	}
+
+	// Watch all the elasticsearch user secrets in the operator namespace. In the future, we may want put this logic in
+	// the utils folder where the other watch logic is.
+	if err = utils.AddSecretWatchWithLabel(c, common.OperatorNamespace(), logstoragecommon.TigeraElasticsearchUserSecretLabel); err != nil {
+		return fmt.Errorf("log-storage-controller failed to watch Secret resource: %w", err)
+	}
+
+	// Watch all the secrets created by this controller so we can regenerate any that are deleted
+	for _, secretName := range []string{
+		render.TigeraElasticsearchGatewaySecret, render.TigeraKibanaCertSecret,
+		render.OIDCSecretName, render.DexObjectName, esmetrics.ElasticsearchMetricsServerTLSSecret,
+		render.TigeraLinseedSecret,
```

Comment:
There are secretwatches that can be added to this loop.

CASecretName is missing